static/app/components/pagination/index.tsx

```diff
@@ -1,97 +0,0 @@
-import {Component, ReactElement} from 'react';
-import {browserHistory} from 'react-router';
-import styled from '@emotion/styled';
-import {Query} from 'history';
-import PropTypes from 'prop-types';
-
-import Button from 'app/components/button';
-import ButtonBar from 'app/components/buttonBar';
-import {IconChevron} from 'app/icons';
-import {t} from 'app/locale';
-import space from 'app/styles/space';
-import {callIfFunction} from 'app/utils/callIfFunction';
-import parseLinkHeader from 'app/utils/parseLinkHeader';
-
-const defaultProps: DefaultProps = {
-  size: 'small',
-  disabled: false,
-  onCursor: (cursor: string, path: string, query: Query, _direction: number) => {
-    browserHistory.push({
-      pathname: path,
-      query: {...query, cursor},
-    });
-  },
-};
-
-type DefaultProps = {
-  size?: 'zero' | 'xsmall' | 'small';
-  onCursor?: (cursor: string, path: string, query: Query, _direction: number) => void;
-  disabled?: boolean;
-};
-
-type Props = DefaultProps & {
-  pageLinks?: string | null;
-  to?: string;
-  /**
-   * The caption must be the PaginationCaption component
```

Comment:
lol didn't even notice this

Reply:
that is what typing is for :trollface:

Reply:
I don't think it's possible to define a specific component, right?

Reply:
maybe....

`React.Element<typeof PaginationCaption>`